### PR TITLE
Add linkTo support for nav.item, deprecate nav.link-to

### LINKED ISSUE #762: Refactor Nav/Item active class support to use Router service
The current hack is ugly, but was necessary when the router service wasn't a thing yet. We can query `isActive()` now, removing the need to observe any child `link-to` components. Also removes the need to have a dedicated yielded `nav/link-to` component extending from `link-to`, so that could be deprecated.

We can do this either now using https://github.com/rwjblue/ember-router-service-polyfill, or wait for the next major bump that drops support for older 2.x versions (preferable).

### PR DESCRIPTION
**Pull request by simonihmig**
Closes #762. 

As Bootstrap expects to have the `active` class on a Nav's `<li>` rather than a child `<a>` (which `{{link-to}}` supports out of the box), the `nav.item` component had to be aware of the active state of its child link. We had to do quite some trickery before: 
* required the use of `nav.link-to` instead of a plain `{{link-to}}`
* this extended `Ember.LinkComponent` and registered itself to the parent `nav.item`
* the `nav.item` then observed the `active` property of its child `nav.link-to` component

Before having Ember's (not so new anymore) router service, I found no better way to do this (with an ergonomic DX), but the implementation is quite awkward (and was error prone in the past, though stable now), in that it queries internal properties of a component (link-to) for some state (is linked route active), that actually lives somewhere else (the router). 

The new proposed API here expects you to set the link params to the `nav.item`, so the component can be aware of its link, and if present will implicitly generate the link by itself, not requiring the use of `nav.link-to` anymore. Uses the Router service (with a polyfill for older versions). The `link-to` component is still around for compatibility reasons, but is being deprecated here and would be removed for 3.0!

Before:

```hbs
{{#nav.item}}{{#nav.link-to "index"}}Home{{/nav.link-to}}{{/nav.item}}
{{#nav.item}}{{#nav.link-to "foo" this.model}}Foo{{/nav.link-to}}{{/nav.item}}
```

After: 

```hbs
{{#nav.item linkTo="index"}}Home{{/nav.item}}
{{#nav.item linkTo=(array "foo" this.model)}}Foo{{/nav.item}}
```

If that's to inflexible for some use case, you can still setup things manually, e.g. by using https://github.com/rwjblue/ember-router-helpers:

```hbs
{{#nav.item active=(is-active "index")}}{{#link-to "index"}}Home{{/link-to}}{{/nav.item}}
```

This was the best solution I could think of, which makes the most common case easy to use without a ton of verbosity. Which does not mean there isn't any possibility for a better one. 🙂 So keeping this open for a while, as this will eventually introduce a breaking change, so folks can voice their thoughts! /cc @jelhan @lolmaus 

PS: this will cause a few merge conflicts with #777 and #778, so will try to get at least #777 merged first, and cleanup the conflicts here...

### CODE REVIEW

File: addon/components/base/bs-nav.js [hunk 1]
```diff
@@ -17,11 +17,11 @@ import layout from 'ember-bootstrap/templates/components/bs-nav';
 
  ```hbs
  {{#bs-nav type="pills" as |nav|}}
-   {{#nav.item}}
-      {{#nav.link-to "foo"}}Foo{{/nav.link-to}}
+   {{#nav.item linkTo="foo"}}Foo
+     Foo
```
Comment (jelhan) [inline addon/components/base/bs-nav.js]: Having `Foo` two times. Should be removed on first line.

File: addon/components/base/bs-nav/link-to.js [hunk 2]
```diff
@@ -12,4 +13,15 @@ import ComponentChild from 'ember-bootstrap/mixins/component-child';
...
 export default LinkComponent.extend(ComponentChild, {
+  init() {
+    this._super(...arguments);
+    deprecate('The link-to component yielded by bs-nav is deprecated. Use the linkTo property of the yielded item component instead!\n' +
+      'Before: {{#nav.item}}{{#nav.link-to "index"}}foo{{/nav.link-to}}{{/nav.item}}\n' +
+      'After: {{#nav.item linkTo="index"}}foo{{/nav.item}}',
+      false,
+      {
+        id: 'ember-bootstrap.nav.link-to',
+        until: '3.0'
```
Comment (jelhan) [inline addon/components/base/bs-nav/link-to.js]: This will be a very short deprecation period isn't it?

Comment (simonihmig) [reply addon/components/base/bs-nav/link-to.js]: Well, who knows? :wink: Don't how quickly all of #780 can be finished...
But 4.0 would be very long I assume, and as this is public API, we can only remove with a major version, right?

**Comment by simonihmig** [comment]
Failing on older versions of Ember, need to investigate this... 🤔

File: addon/components/base/bs-nav/item.js [hunk 3]
```diff
@@ -22,6 +24,22 @@ export default Component.extend(ComponentParent, {
   classNameBindings: ['disabled', 'active'],
   tagName: 'li',
   ariaRole: 'presentation',
+  router: service(),
+
+  /**
+   * If set will wrap the item's content in a link. Accepts either a string with the route name, or an array
+   * with the route name and one or multiple models / query params, similar to the positional params of `{{link-to ...}}`
```
Comment (jelhan) [inline addon/components/base/bs-nav/item.js]: Maybe docs should also mention query params?

Comment (simonihmig) [reply addon/components/base/bs-nav/item.js]: It (briefly) does: "with the route name and one or multiple models / query params". Would you expect/suggest something more verbose?

Comment (jelhan) [reply addon/components/base/bs-nav/item.js]: Oh, of course. 😧 Sorry for the noice.

**Pull request merged**